File: chk_kc.cpp
// Tool to check kyotocabinet db
// Build: g++ -o chk_kc -lkyotocabinet chk_kc.cpp

#include <ostream>
#include <kcpolydb.h>

using namespace std;

bool chk_one(const string dir, const string name) {
  kyotocabinet::PolyDB db;

  auto path = dir + "/" + name + ".kch";
  if (!db.open(path, kyotocabinet::PolyDB::OREADER)) {
    cerr << "Can't open db '" << path << "'." << endl;
    return false;
  }
  cout << name << ": " << db.count() << endl;
  return true;
}

int main(int argc, char *argv[]) {
  if (argc != 2) {
    cerr << "Usage: " << argv[0] << " <k-v dir>" << endl;
    return 1;
  }
  if (chk_one(argv[1], "tx"))
    chk_one(argv[1], "addr");
  return 0;
}

File: chk_kc.py
#!/bin/env python3
"""
Tool to check kyotocabinet storages
"""

import sys, os
import kyotocabinet

def chk_one (dir: str, name: str) -> bool:
  """ Check one db """
  fname = os.path.join(dir, name+'.kch')
  if not os.path.isfile(fname):
    print("'{}' not exist or is not file.".format(fname))
    return False
  db = kyotocabinet.DB(kyotocabinet.DB.GEXCEPTIONAL)
  if not db.open(fname, kyotocabinet.DB.OREADER):
    print("Can't open db '{}'.".format(fname))
    return False
  print("{}: {} records".format(name, db.count()))
  return True

if __name__ == '__main__':
  if len(sys.argv) != 2:
    print("Usage: {} <k-v dir>".format(sys.argv[0]))
  else:
    dir = sys.argv[1]
    if not os.path.isdir(dir):
      print("'{}' not exist or is not folder.".format(dir))
    else:
      if chk_one(dir, 'tx'):
        chk_one(dir, 'addr')
